src/EnergyPlus/ExtendedHI.cc

```diff
@@ -0,0 +1,538 @@
+// CONTRIBUTORS BE LIABLE FOR ANY DIRECT, INDIRECT, INCIDENTAL, SPECIAL, EXEMPLARY, OR
+// CONSEQUENTIAL DAMAGES (INCLUDING, BUT NOT LIMITED TO, PROCUREMENT OF SUBSTITUTE GOODS OR
+// SERVICES; LOSS OF USE, DATA, OR PROFITS; OR BUSINESS INTERRUPTION) HOWEVER CAUSED AND ON ANY
+// THEORY OF LIABILITY, WHETHER IN CONTRACT, STRICT LIABILITY, OR TORT (INCLUDING NEGLIGENCE OR
+// OTHERWISE) ARISING IN ANY WAY OUT OF THE USE OF THIS SOFTWARE, EVEN IF ADVISED OF THE
+// POSSIBILITY OF SUCH DAMAGE.
+
+// Adapted from the code Version 1.1 released by Yi-Chuan Lu on February 23, 2023.
+//
+// @article{20heatindex,
+//   Title   = {Extending the Heat Index},
+//   Author  = {Yi-Chuan Lu and David M. Romps},
+//   Journal = {Journal of Applied Meteorology and Climatology},
+//   Year    = {2022},
+//   Volume  = {61},
+//   Number  = {10},
+//   Pages   = {1367--1383},
+//   Year    = {2022},
+// }
+//
+
+#include <cmath>
+
+#include <EnergyPlus/Data/EnergyPlusData.hh>
+#include <EnergyPlus/ExtendedHI.hh>
+#include <EnergyPlus/General.hh>
+#include <EnergyPlus/HVACSystemRootFindingAlgorithm.hh>
+
+namespace EnergyPlus {
+
+namespace ExtendedHI {
+    // Thermodynamic parameters
+    constexpr Real64 Ttrip = 273.16; // K
+    constexpr Real64 ptrip = 611.65; // Pa
+    constexpr Real64 E0v = 2.3740e6; // J/kg
+    constexpr Real64 E0s = 0.3337e6; // J/kg
+    constexpr Real64 rgasa = 287.04; // J/kg/K
+    constexpr Real64 rgasv = 461.;   // J/kg/K
+    constexpr Real64 cva = 719.;     // J/kg/K
+    constexpr Real64 cvv = 1418.;    // J/kg/K
+    constexpr Real64 cvl = 4119.;    // J/kg/K
+    constexpr Real64 cvs = 1861.;    // J/kg/K
+    constexpr Real64 cpa = cva + rgasa;
+    constexpr Real64 cpv = cvv + rgasv;
+
+    // The saturation vapor pressure
+    Real64 pvstar(Real64 T)
+    {
+        if (T == 0.0) {
+            return 0.0;
+        } else if (T < Ttrip) {
+            return ptrip * pow((T / Ttrip), ((cpv - cvs) / rgasv)) * exp((E0v + E0s - (cvv - cvs) * Ttrip) / rgasv * (1. / Ttrip - 1. / T));
+        } else {
+            return ptrip * pow((T / Ttrip), ((cpv - cvl) / rgasv)) * exp((E0v - (cvv - cvl) * Ttrip) / rgasv * (1. / Ttrip - 1. / T));
+        }
+        return 0.0;
+    }
+
+    // The latent heat of vaporization of water
+    Real64 Le(Real64 T)
+    {
+        return (E0v + (cvv - cvl) * (T - Ttrip) + rgasv * T);
+    }
+
+    // Thermoregulatory parameters
+    constexpr Real64 sigma = 5.67e-8;                 // W/m^2/K^4 , Stefan-Boltzmann constant
+    constexpr Real64 epsilon = 0.97;                  //           , emissivity of surface, steadman1979
+    constexpr Real64 M = 83.6;                        // kg        , mass of average US adults, fryar2018
+    constexpr Real64 H = 1.69;                        // m         , height of average US adults, fryar2018
+    Real64 A = 0.202 * pow(M, 0.425) * pow(H, 0.725); // m^2       , DuBois formula, parson2014
```

Comment:
This should be constexpr-able right?  And I will assume for now that these are indeed used in multiple locations where it makes sense to keep them out here in the namespace.

Reply:
yes. It should be. Yeah, some stuff like `Pc` is used in a few functions.